src/utils/transaction/userOperation.ts

```diff
@@ -43,12 +45,31 @@ export const generateSignedUserOperation = async (
     address: accountAddress as Address,
   });
 
+  let accountFactoryAddress: Address | undefined;
+
+  try {
+    const cachedFactoryAddress = await redis.get(
+      `account-factory:${accountAddress.toLowerCase()}`,
+    );
+    accountFactoryAddress = getAddress(cachedFactoryAddress ?? "");
+  } catch {}
+
   // Resolve Factory Contract Address from Smart-Account Contract
-  const accountFactoryAddress = await readContract({
-    contract: smartAccountContract,
-    method: "function factory() view returns (address)",
-    params: [],
-  });
+
+  if (!accountFactoryAddress) {
+    const onchainFactoryResult = (await readContract({
```

Comment:
Unless I'm missing something, shouldn't you just set the response directly to `accountFactoryAddress` instead of declaring a new var?